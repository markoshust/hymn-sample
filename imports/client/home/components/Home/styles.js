const styles = {
  link: {
    display: 'block',
  },
};

export default styles;
