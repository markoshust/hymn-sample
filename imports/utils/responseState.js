const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
