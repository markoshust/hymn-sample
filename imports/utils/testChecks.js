import renderer from 'react-test-renderer';

const componentCheck = (component) => {
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

const renderCheck = (children) => {
  const component = renderer.create(children);

  return componentCheck(component);
};

export {
  componentCheck,
  renderCheck,
};
