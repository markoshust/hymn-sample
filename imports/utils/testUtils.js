import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
