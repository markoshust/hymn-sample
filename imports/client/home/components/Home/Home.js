import React from 'react';
import { compose, pure } from 'recompose';
import { Link } from 'react-router';
import useSheet from 'react-jss';
import styles from './styles';

const Home = () => (
  <div>
    <h1>Home</h1>
    <nav>
      <Link to="/welcome" style={styles.link}>Welcome to Meteor</Link>
    </nav>
  </div>
);

const enhance = compose(
  useSheet(styles),
  pure,
);

export default enhance(Home);
