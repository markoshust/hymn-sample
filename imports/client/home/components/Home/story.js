import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Home from './Home';

const stories = storiesOf('home/Home', module);

stories.add('Home', () => (
  <Home />
));
