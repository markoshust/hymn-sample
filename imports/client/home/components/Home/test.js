import React from 'react';
import { renderCheck } from '/imports/utils/testChecks';
import Home from './Home';

describe('<Home />', () => {
  it('renders correctly', () => {
    renderCheck(<Home />);
  });
});
