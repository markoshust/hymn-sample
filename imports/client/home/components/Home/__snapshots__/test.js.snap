exports[`<Home /> renders correctly 1`] = `
<div>
  <h1>
    Home
  </h1>
  <nav>
    <a
      onClick={[Function bound handleClick]}
      style={
        Object {
          "display": "block"
        }
      }>
      Welcome to Meteor
    </a>
  </nav>
</div>
`;
